torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -2591,7 +2591,7 @@ c10::intrusive_ptr<Work> ProcessGroupNCCL::collective(
     work->stashed_for_allocator_safety_->push_back(input);
   }
 
-  at::cuda::OptionalCUDAGuard gpuGuard;
```

Comment:
Is it the case that gpuGuard without index is not effective? Or does it default to device 0?